codec-http/src/test/java/io/netty/handler/codec/http/multipart/AbstractMemoryHttpDataTest.java

```diff
@@ -59,6 +62,36 @@ public void testSetContentFromStream() throws Exception {
         }
     }
 
+    @Test
+    public void testSetContentFromStreamExceptionally() throws Exception {
+        TestHttpData data = new TestHttpData("test", UTF_8, 0); //definedSize=0
+        data.setMaxSize(10);
+        byte[] bytes = new byte[16];
+        UnpooledByteBufAllocator alloc = UnpooledByteBufAllocator.DEFAULT;
```

Comment:
Create a new instance of `UnpooledByteBufAllocator` as otherwise this may break if tests are executed concurrently.

Reply:
Creating a new instance of UnpooledByteBufAllocator is a good idea to handle the concurrent test cases. And we must make the temporary ByteBuf instance be allocated from this new allocator as well. But I found that buffer method called by setContent method is statically imported. 
https://github.com/netty/netty/blob/81513c3728df8add3c94fd0bdaaf9ba424925b29/codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractMemoryHttpData.java#L65-L69
https://github.com/netty/netty/blob/81513c3728df8add3c94fd0bdaaf9ba424925b29/codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractMemoryHttpData.java#L31-L34
Maybe I can extract it to a new protected method and override it so that I can make the ByteBuf be allocate from my new allocator. However, for test purpose, I'm not sure whether it's worthwhile. @normanmaurer Any suggestions?

Reply:
just remove all of this... the leak detection will kick in if we leak the buffer anyway.